src/Core/tests/UnitTests/Layouts/GridLayoutManagerTests.cs

```diff
@@ -63,6 +64,41 @@ IGridLayout CreateGridLayout(int rowSpacing = 0, int colSpacing = 0,
 			return grid;
 		}
 
+		IGridLayout CreateGridLayoutWithDensity(double density, int rowSpacing = 0, int colSpacing = 0,
```

Comment:
The implementation here isn't correct. 

Make it so `IView` implements `IViewWithWindow` and then you can just setup a Substitue `IWindow` that gets returned from `IGridLayout`

And then you don't need to setup all these other substitutes